spiffworkflow-backend/src/spiffworkflow_backend/models/json_data.py

```diff
@@ -67,7 +67,7 @@ def insert_or_update_json_data_records(cls, json_data_hash_to_json_data_dict_map
             on_duplicate_key_stmt = None
             if current_app.config["SPIFFWORKFLOW_BACKEND_DATABASE_TYPE"] == "mysql":
                 insert_stmt = mysql_insert(JsonDataModel).values(list_of_dicts)
-                on_duplicate_key_stmt = insert_stmt.on_duplicate_key_update(data=insert_stmt.inserted.data)
+                on_duplicate_key_stmt = insert_stmt.on_duplicate_key_update(hash=insert_stmt.inserted.hash)
```

Comment:
Using insert on duplicate update should ensure that we allow all writes to succeed, rather than having one duplicate key failure prevent writing any data. We do not really care what happens with the update statement beyond that, since whether we update the value or the hash, it should never have any effect, since for any given hash we always write the same data.